python/plugins/processing/algs/qgis/SimplifyGeometries.py

```diff
@@ -58,20 +59,21 @@ def group(self):
 
     def __init__(self):
         super().__init__()
-        self.addParameter(ParameterVector(self.INPUT,
-                                          self.tr('Input layer'),
-                                          [dataobjects.TYPE_VECTOR_POLYGON, dataobjects.TYPE_VECTOR_LINE]))
+        self.addParameter(QgsProcessingParameterFeatureSource(self.INPUT,
+                                                              self.tr('Input layer'),
+                                                              [dataobjects.TYPE_VECTOR_POLYGON, dataobjects.TYPE_VECTOR_LINE]))
         self.methods = [self.tr('Distance (Douglas-Peucker)'),
                         'Snap to grid',
                         'Area (Visvalingam)']
-        self.addParameter(ParameterSelection(
+        self.addParameter(QgsProcessingParameterEnum(
             self.METHOD,
             self.tr('Simplification method'),
-            self.methods, default=0))
-        self.addParameter(ParameterNumber(self.TOLERANCE,
-                                          self.tr('Tolerance'), 0.0, 10000000.0, 1.0))
+            self.methods, defaultValue=0))
+        self.addParameter(QgsProcessingParameterNumber(self.TOLERANCE,
+                                                       self.tr('Tolerance'), minValue=0.0, maxValue=10000000.0, defaultValue=1.0))
 
-        self.addOutput(OutputVector(self.OUTPUT, self.tr('Simplified')))
+        self.addParameter(QgsProcessingParameterFeatureSink(self.OUTPUT, self.tr('Simplified')))
+        self.addOutput(QgsProcessingOutputVectorLayer(self.OUTPUT, self.tr('Simplified')))
```

Comment:
Ok - so the feature sink parameter is the *input* specifying where the resulting layer should be created. That's the important bit.

Outputs are now used solely for indicating what generated values are available after running the alg within a model. So the algs need to report that they generate a vector layer, which can then be used as a input to other child algs later in a model.

There's a chance that some algs will create layers *without* giving users an option to specify where those layers are to be created, in which only the output will be set. This may be the case when a alg outputs multiple layers, or is hardcoded to place an output layer in a specific location. 

Similarly, there's a chance that algs will be written which generate layers, but (for whatever reason) don't want these layers available for use in models.

My intention here is that algs will be soaked in outputs. Things like outputing the count of features processed, etc. This would mean all those useful outputs become available for use in model logic.